---
title: "Lab Notebook"
author: "EShelton"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# First Steps

My directory is found at: /projectnb/vervpop/eshelton. (Input: "cd /projectnb/vervpop" "cd eshelton")

The location of *PPARGC1A*, or *PGC-1Î±*, in the Vervet genome is **27: 26,465,499-26,564,582**. With a buffer zone of 10k base pairs, the location is **27: 26,455,499-26,574,582**.

## Getting the gene region:

```{r}`r''`
wget https://vervetmonkeygmi.blob.core.windows.net/publicationdata/163_201701_UG_vervet_monkey_SNPs_all_chrom_beagle_shapeit.vcf.gz

tabix -p vcf 163_201701_UG_vervet_monkey_SNPs_all_chrom_beagle_shapeit.vcf.gz.1

tabix -h 163_201701_UG_vervet_monkey_SNPs_all_chrom_beagle_shapeit.vcf.gz.1 CAE27:26455499-26574582 > vervPGC1a.full.vcf

less vervPGC1a.full.vcf
```
```
##fileformat=VCFv4.1
##FILTER=<ID=PASS,Description="All filters passed">
##fileDate=14022017_09h11m56s
##source=SHAPEIT2.v837
##log_file=shapeit_14022017_09h11m56s_34f53654-f41e-4bf1-89fe-892d6199343a.log
##FORMAT=<ID=GT,Number=1,Type=String,Description="Phased Genotype">
##contig=<ID=CAE1>
##contig=<ID=CAE2>
##contig=<ID=CAE3>
##contig=<ID=CAE4>
##contig=<ID=CAE5>
##contig=<ID=CAE6>
##contig=<ID=CAE7>
##contig=<ID=CAE8>
##contig=<ID=CAE9>
##contig=<ID=CAE10>
##contig=<ID=CAE11>
##contig=<ID=CAE12>
##contig=<ID=CAE13>
##contig=<ID=CAE14>
##contig=<ID=CAE15>
##contig=<ID=CAE16>
##contig=<ID=CAE17>
##contig=<ID=CAE18>
vervPGC1a.full.vcf
```
Hit 'q' to close the less command view.
```{r}`r''`
wc -l vervPGC1a.full.vcf
```
```
2340 vervPGC1a.full.vcf
```
```{r}`r''`
grep -c "^##" vervPGC1a.full.vcf
```
```
38  
```

Note: At this point vervPGC1a.full.vcf was renamed to vervpcg1a.full.vcf.

Questions:  
If the download of a file from the SCC is interrupted accidentally, how do I unpause the download without starting it all over again?     
When I used the 'tabix -p' command, when I hit return it just went down a line; after a couple minutes it goes. How do I make it run sooner?  

## Functional Regions

Number: Start coordinate-End coordinate (length)

**Exon 1: 26,465,499-26,465,783 (285)**  
  *Intron 1-2: 26465784-26470781 (4998)*   
**Exon 2: 26470782-26470961 (180)**   
  *Intron 2-3: 26470962-26523412 (52451)*   
**Exon 3: 26523413-26523607 (195)**   
  *Intron 3-4: 26523608-26525582 (1975)*   
**Exon 4: 26525583-26525705 (123)**   
  *Intron 4-5: 26525706-26526565 (860)*   
**Exon 5: 26526566-26526770 (205)**   
  *Intron 5-6: 26526771-26530747 (3977)*   
**Exon 6: 26530748-26530793 (46)**   
  *Intron 6-7: 26530794-26530902 (109)*   
**Exon 7: 26530903-26530976 (74)**   
  *Intron 7-8: 26530977-26540652 (9676)*   
**Exon 8: 26540653-26541568 (916)**    
  *Intron 8-9: 26541569-26542129 (561)*   
**Exon 9: 26542130-26542234 (105)**   
  *Intron 9-10: 26542235-26542387 (153)*   
**Exon 10: 26542388-26542508 (121)**   
  *Intron 10-11: 26542509-26552948 (10440)*   
**Exon 11: 25552949-26553070 (122)**   
  *Intron 11-12: 26553071-26553415 (345)*   
**Exon 12: 26553416-26553567 (152)**   
  *Intron 12-13: 26553568-26559651 (6084)*   
**Exon 13: 26559652-26564582 (4931)**

## Hardy-Weinburg Equilibrium

I ran into trouble with the 'vervet.population.panel' file required for this section. It wasn't in the vervpop directory, so Christian Gagnon sent it to me and I downloaded it onto my computer. I then had issues with uploading the file to the SCC. The SCC cheat sheet pdf used the command 'scp *filename* *username*@scc1.bu.edu:~', but when I ran that command the terminal said that no such file or directory existed. I tried it in my home directory, the vervpop directory, and my eshelton directory. I was able to upload the file using SCC On Demand.

I created the file 'pygpgc1a.vcf' of the variants within the South African *pygerythrus* sample using the given code, deleting the VSAI3005 sample. 

```{r}`r''`
grep pygerythrus vervet.population.panel | cut -f1 > pyg.samples.list
gedit pyg.samples.list
module load vcftools
vcftools --vcf vervUCP1.full.vcf --keep pyg.samples.list --recode --out pygUCP1
mv pygUCP1.recode.vcf pygUCP1.vcf
```

I created the file 'sepgc1a.vcf' of the variants within the Southern Expansion using the given code. I left the VCSAI3005 sample in. The instructions also mentioned changing the taxonomic designation of two of the samples, which I was unsure how to do.

# SNPs

I opened an SCC OnDemand R server, and used the R code given in the pipeline to analyze the 'pygpgc1a.hwe' file. It returned 102 SNPs. The 'sepgc1a.hwe' file returned 276 SNPs. The details of these have been copied to text files. 

There was a complication and Christian Gagnon sent me the correct version of the 'vervet.population.panel' file. I redid the creation of the Pygerythrus and Southern Expansion files, 'pygpgc1a.hwe' and 'sepgc1a.hwe', making sure to edit 'vervet.population.panel' file in between to make the changes necessary to the VBOA and VSAJ2008 samples.

The new 'pygpgc1a.hwe' and 'sepgc1a.hwe' files returned 102 SNPs each. The files have a lot of SNPs in common.

# Linkage Disequilibrium

Following the pipeline instructions, I created the 'pygpgc1a_loci.txt' and 'sepgc1a_loci.txt' files in the directory, then opened an R server and installed the necessary packages. 
```{r}`r''`
library(devtools)
library(RMariaDB)
library(BiocManager)
devtools::install_github("SFUStatgen/LDheatmap")
BiocManager::install("snpStats")
BiocManager::install("VariantAnnotation")
BiocManager::install("GenomicFeatures")
BiocManager::install("gpart")
library(tidyverse)
library(snpStats)
library(VariantAnnotation)
library(LDheatmap)
library(GenomicFeatures)
library(gpart)
library(Matrix)
```


### LD: *pygerythrus*

```{r}`r''`
> pygpgc1a_vcf<-readVcf("pygpgc1a.vcf")
> pygpgc1a_matrix<-genotypeToSnpMatrix(pygpgc1a_vcf)
> pygpgc1a_vcf
```
```
class: CollapsedVCF 
dim: 2301 48 
rowRanges(vcf):
  GRanges with 5 metadata columns: paramRangeID, REF, ALT, QUAL, FILTER
info(vcf):
  DataFrame with 0 columns: 
geno(vcf):
  List of length 1: GT
geno(header(vcf)):
      Number Type   Description    
   GT 1      String Phased Genotype
```

I generated an LD heatmap for *pygerythrus*.

```{r}`r''`
> cols=colorRampPalette(c("yellow","red"))(10)
> image(pygpgc1a_LD,lwd=0,cuts=9,col.regions=cols,colorkey=TRUE)
```


I then generated a more complex heatmap, by linking the SNPs with the vervet references.

```{r}`r''`
> vcfbig<-readVcf("pygpgc1a.vcf","txdb")
> ALLmatrix<-genotypeToSnpMatrix(vcfbig)
> ALLmatrix
```
```
$genotypes
A SnpMatrix with  48 rows and  2301 columns
Row names:  VSAA2010 ... VSAM5007 
Col names:  CAE27:26455516_G/A ... CAE27:26574535_G/A 

$map
DataFrame with 2301 rows and 4 columns
              snp.names       allele.1           allele.2    ignore
            <character> <DNAStringSet> <DNAStringSetList> <logical>
1    CAE27:26455516_G/A              G                  A     FALSE
2    CAE27:26455526_G/A              G                  A     FALSE
3    CAE27:26455669_C/T              C                  T     FALSE
4    CAE27:26456003_C/T              C                  T     FALSE
5    CAE27:26456070_G/A              G                  A     FALSE
...                 ...            ...                ...       ...
2297 CAE27:26574262_C/G              C                  G     FALSE
2298 CAE27:26574290_T/C              T                  C     FALSE
2299 CAE27:26574462_C/A              C                  A     FALSE
2300 CAE27:26574533_C/T              C                  T     FALSE
2301 CAE27:26574535_G/A              G                  A     FALSE
```
```{r}`r''`
> allheatmapr <- LDheatmap(ALLmatrix$genotypes,
+                          genetic.distances=positions_pygpgc1a,
+                         distances="physical",LDmeasure="r",
+                          title="Pairwise LD with R^2",
+                          add.map=TRUE, add.key=TRUE,
+                          geneMapLocation=0.15,
+                         SNP.name=c("CAE27:26457890_C/G","CAE27:26512806_G/A","CAE27:26512817_A/C","CAE27:26512838_A/G","CAE27:26512856_G/A","CAE27:26512884_C/G","CAE27:26522930_G/A","CAE27:26522937_G/A","CAE27:26522971_G/A","CAE27:26522990_A/G","CAE27:26522992_C/T","CAE27:26522994_A/G","CAE27:26549769_G/T","CAE27:26558338_T/C","CAE27:26558347_G/T","CAE27:26573010_T/C"),
+                          color=NULL, newpage=TRUE,
+                          name="ldheatmap")
```
I was unsure which SNPs to highlight, so I chose the first, the last, and several SNPS that were located closely together.


The {gpart} code found five linkage blocks.
```{r}`r''`
> pyg_pgc1a_res1 = BigLD(genofile = "pygpgc1a.vcf", LD="r2")
> pyg_pgc1a_res1
```
```
    chr start.index end.index start.rsID end.rsID start.bp   end.bp
1 CAE27           8       786          .        . 26456233 26495925
2 CAE27         804      2205          .        . 26497136 26568734
3 CAE27        2214      2283          .        . 26569080 26573306
4 CAE27        2285      2288          .        . 26573416 26573643
5 CAE27        2291      2292          .        . 26573987 26573990
```
```{r}`r''`
> LDblockHeatmap(genofile = "pygpgc1a.vcf", chrN = "CAE27", geneDB = "ensembl", geneid = "hgnc_symbol", blocktype = "bigld", LD = "r2", type = "png", filename = "heatmap_pygpgc1a_r2", CLQmode = "density", res = 600)
```   

The 'heatmap_pygpgc1a_r2.png' file is in the PGC-1a directory.

The specific SNPs are the hwe coordinates with the "_X/x", in the ALLmatrix and ALLmatrix2. 

### LD: Southern Expansion

```{r}`r''`
> sepgc1a_vcf<-readVcf("sepgc1a.vcf")
> sepgc1a_matrix<-genotypeToSnpMatrix(sepgc1a_vcf)
> sepgc1a_vcf
```
```
class: CollapsedVCF 
dim: 2301 73 
rowRanges(vcf):
  GRanges with 5 metadata columns: paramRangeID, REF, ALT, QUAL, FILTER
info(vcf):
  DataFrame with 0 columns: 
geno(vcf):
  List of length 1: GT
geno(header(vcf)):
      Number Type   Description    
   GT 1      String Phased Genotype
```
```{r}`r''`
> sepgc1a_LD<-ld(sepgc1a_matrix$genotypes, depth = 825, stats = "R.squared")
> cols=colorRampPalette(c("yellow","red"))(10)
> image(sepgc1a_LD, lwd=0, cuts=9, col.regions=cols, colorkey = TRUE)
```
```{r}`r''`
> txdb<-makeTxDbFromEnsembl(organism = "Chlorocebus sabaeus", server = "useastdb.ensembl.org")
> vcfbig2<-readVcf("sepgc1a.vcf", "txdb")
> ALLmatrix2<-genotypeToSnpMatrix(vcfbig2)
> ALLmatrix2
```
```{r}`r''`
> allheatmapr2 <- LDheatmap(ALLmatrix2$genotypes,
+                           genetic.distances=positions_sepgc1a,
+                           distances="physical",LDmeasure="r",
+                           title="Pairwise LD with R^2",
+                           add.map=TRUE, add.key=TRUE,
+                           geneMapLocation=0.15,
+                         SNP.name=c("CAE27:26457890_C/G","CAE27:26512806_G/A","CAE27:26512817_A/C","CAE27:26512838_A/G","CAE27:26512856_G/A","CAE27:26512884_C/G","CAE27:26522930_G/A","CAE27:26522937_G/A","CAE27:26522971_G/A","CAE27:26522990_A/G","CAE27:26522992_C/T","CAE27:26522994_A/G","CAE27:26549769_G/T","CAE27:26558338_T/C","CAE27:26558347_G/T","CAE27:26573010_T/C"),
+                         color=NULL, newpage=TRUE,                         name="ldheatmap")
```

```
$genotypes
A SnpMatrix with  73 rows and  2301 columns
Row names:  AG23 ... VZC1020 
Col names:  CAE27:26455516_G/A ... CAE27:26574535_G/A 

$map
DataFrame with 2301 rows and 4 columns
              snp.names       allele.1           allele.2    ignore
            <character> <DNAStringSet> <DNAStringSetList> <logical>
1    CAE27:26455516_G/A              G                  A     FALSE
2    CAE27:26455526_G/A              G                  A     FALSE
3    CAE27:26455669_C/T              C                  T     FALSE
4    CAE27:26456003_C/T              C                  T     FALSE
5    CAE27:26456070_G/A              G                  A     FALSE
...                 ...            ...                ...       ...
2297 CAE27:26574262_C/G              C                  G     FALSE
2298 CAE27:26574290_T/C              T                  C     FALSE
2299 CAE27:26574462_C/A              C                  A     FALSE
2300 CAE27:26574533_C/T              C                  T     FALSE
2301 CAE27:26574535_G/A              G                  A     FALSE
```

The {gpart} code found seven linkage blocks.

```{r}`r''`
> se_pgc1a_res1 = BigLD(genofile = "sepgc1a.vcf", LD = "r2")
> se_pgc1a_res1
```
```
    chr start.index end.index start.rsID end.rsID start.bp   end.bp
1 CAE27          11      1327          .        . 26456426 26517910
2 CAE27        1338      1356          .        . 26518801 26519695
3 CAE27        1361      1391          .        . 26520054 26521556
4 CAE27        1408      1437          .        . 26522252 26522994
5 CAE27        1438      1447          .        . 26523036 26523284
6 CAE27        1451      1509          .        . 26523595 26526968
7 CAE27        1514      2301          .        . 26527279 26574535
```
```{r}`r''`
> LDblockHeatmap(genofile = "sepgc1a.vcf", chrN = "CAE27", geneDB = "ensembl", geneid = "hgnc_symbol", blocktype = "bigld", LD = "r2", type = "png", filename = "heatmap_sepgc1a_r2", CLQmode = "density", res = 600)
```

The 'heatmap_sepgc1a_r2.png' file can be found in the PGC-1a directory.

